### ISSUE TITLE
TCPSite name property returns incorrect URL when ephemeral ports are used

### ISSUE THREAD

**Issue by claws** [aio-libs/aiohttp#3018]
## Long story short

The ``aiohttp.web.TCPSite`` object provides a ``name`` property allowing a developer to access the URL provided by the site. When using ephemeral ports the URL returned is wrong.

### Long story

After upgrading to aiohttp 3.2.1 I observed the following deprecation warnings:
```
DeprecationWarning: Application.make_handler(...) is deprecated, use AppRunner API instead
```
I investigated AppRunner API and modified my code accordingly to upgrade it. I then observed errors in unit tests that were related to the requesting aiohttp to use ephemeral ports for service endpoints.

In my Python async applications one or more aiohttp servers are running within an application to provide services such as application metrics, diagnostics, etc. Many applications run on a node (no containers here). Ephemeral ports are used to avoid port conflicts between application services and a service discovery mechanism is used to help resolve ephemeral services. I also use ephemeral ports in unit tests to avoid port conflicts with system service ports and other CI jobs running in parallel.

After starting aiohttp web services within my application I fetch their actual URL and register it with a service discovery mechanism. The URL returned from aiohttp is wrong.

## Expected behaviour

Accessing the ``name`` property of a started ``TCPSite`` object returns the correct URL string.

## Actual behaviour

Accessing the ``name`` property of a started ``TCPSite`` object returns the incorrect port in the URL string when ephemeral ports are used.

## Steps to reproduce

Starting a TCPSite with an ephemeral port (port=0) can be used to reproduce the problem. The following script demonstrates this.
``` python
import asyncio
import aiohttp
import aiohttp.web
from yarl import URL


async def demo():
    addr = "127.0.0.1"
    port = 0
    ssl = None

    # Start web site.
    myapp = aiohttp.web.Application()
    runner = aiohttp.web.AppRunner(myapp)
    await runner.setup()

    site = aiohttp.web.TCPSite(runner, addr, port, ssl_context=ssl)
    await site.start()

    # Service is now operational - fetch url to it can be registered with service discovery.

    # fetch URL using aiohttp API call
    site_url_api = site.name

    # fetch URL using my work-around that gets the actual port in use.
    # NOTE: getsockname => IPv4 returns 2-tuple, IPv6 returns 4-tuple
    actual_host, actual_port, *_ = site._server.sockets[0].getsockname()
    site_url_workaround = str(
        URL.build(
            scheme='https' if ssl is not None else 'http',
            host=actual_host if ":" not in actual_host else "[{}]",  # handle IPv4 and IPv6
            port=actual_port))

    # Check if aiohttp returns the correct URL
    if site_url_api != site_url_workaround:
        print(f'ERROR: aiohttp url: {site_url_api} != actual url: {site_url_workaround}')

    # Stop web site.
    await site.stop()
    await runner.cleanup()
    loop = asyncio.get_event_loop()
    await loop.shutdown_asyncgens()


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(demo())

```

When I run the script I observe:
```
ERROR: aiohttp url: http://127.0.0.1 != actual url: http://127.0.0.1:54804
```
## Your environment

Python3.6
aiohttp==3.2.1
Mac OSX
'server'

**Comment by asvetlov** [comment]
Not sure if I want to support ephemeral ports at all.
Now there is no way to get the port back.
`.name` is *informational* property, not URL. The format may be changed without any notification. Consider it as `__repr__`.

**Comment by asvetlov** [comment]
Solved by #3037
